src/Halcyon/MemoryRepository.php

```diff
@@ -0,0 +1,172 @@
+<?php namespace October\Rain\Halcyon;
+
+use Illuminate\Cache\Repository;
+
+/**
+ * Provides a simple request-level cache.
+ *
+ * @package october\halcyon
+ * @author Alexey Bobkov, Samuel Georges
+ */
+class MemoryRepository extends Repository
+{
+    /**
+     * Values stored in memory
+     *
+     * @var array
+     */
+    protected $cache = [];
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param  string|array $key
+     * @return mixed
+     */
+    public function get($key, $default = null)
+    {
+        if(is_array($key)) {
+            return $this->many($key);
+        }
+
+        $value = $this->getFromMemoryCache($key);
+
+        if(!is_null($value)) {
+            return $value;
+        }
+
+        $value = parent::get($key, $default);
+        $this->putInMemoryCache($key, $value);
+
+        return $value;
+    }
+
+    /**
+     * Store an item in the cache.
+     *
+     * @param  string  $key
+     * @param  mixed   $value
+     * @param  \DateTimeInterface|\DateInterval|float|int  $minutes
+     * @return void
+     */
+    public function put($key, $value, $minutes = null)
+    {
+        if (is_array($key)) {
+            $this->putMany($key, $value);
```

Comment:
See https://github.com/laravel/framework/issues/23729 for a reason why `is_array` check should be removed entirely here.

Reply:
@taylorotwell has decided to not remove it (even though it's not documented anywhere and the argument mapping doesn't line up) so we'll have to leave it here for compatibility with Laravel.

Reply:
Indeed, I agree it looks very strange, although I guess it technically works. You'd have to call it like `put(['foo' => 'bar'], 10)` with the `$value` param effectively being used as `$minutes`.